src/Tribe/Settings_Tab.php

```diff
@@ -349,6 +366,147 @@ public function get_priority(): string {
+
+		// Determine the default value.
+		$default = $field['default'] ?? null;
+
+		/**
+		 * Filter the default value of the field.
+		 *
+		 * @param mixed $default The default value of the field.
+		 * @param array $field   The field array.
+		 */
+		$default = apply_filters( 'tribe_settings_field_default', $default, $field );
+
+		// If there's no parent option, get the site option (for network admin) or the option.
+		if ( ! $parent_option ) {
+			return ( $network_option || is_network_admin() )
+				? get_site_option( $key, $default )
+				: get_option( $key, $default );
+		}
+
+		// Get the options from Tribe__Settings_Manager if we're getting the main array.
+		if ( $parent_option === Tribe__Main::OPTIONNAME ) {
+			return Tribe__Settings_Manager::get_option( $key, $default );
+		}
+
+		// Get the network options from Tribe__Settings_Manager.
+		if ( $parent_option === Tribe__Main::OPTIONNAMENETWORK ) {
+			return Tribe__Settings_Manager::get_network_option( $key, $default );
+		}
+
+		// Get the parent option for network admin.
+		if ( is_network_admin() ) {
+			$options = (array) get_site_option( $parent_option );
+
+			return $options[ $key ] ?? $default;
+		}
+
+		// Else, get the parent option normally.
+		$options = (array) get_option( $parent_option );
+
+		return $options[ $key ] ?? $default;
+	}
+
+	/**
+	 * Adds a sidebar to the current tab.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function has_sidebar(): bool {
+		return null !== $this->sidebar;
+	}
+
+	/**
+	 * Gets the sidebar for the current tab.
+	 *
+	 * @since TBD
+	 *
+	 * @return Settings_Sidebar|null
+	 * @throws InvalidArgumentException If the sidebar is invalid.
+	 */
+	public function get_sidebar() {
+		if ( $this->sidebar instanceof Settings_Sidebar ) {
+			return $this->sidebar;
+		}
+
+		if ( is_callable( $this->sidebar ) ) {
+			$sidebar = call_user_func( $this->sidebar );
+			if ( ! $sidebar instanceof Settings_Sidebar ) {
+				throw new InvalidArgumentException(
+					esc_html__( 'The sidebar callback must return an instance of Settings_Sidebar', 'tribe-common' )
+				);
+			}
+
+			return $sidebar;
+		}
+
+		return null;
+	}
+
+	/**
+	 * Sets the sidebar for the current tab.
+	 *
+	 * @param callable|Settings_Sidebar $sidebar The sidebar to set.
+	 *
+	 * @return void
+	 */
+	public function add_sidebar( $sidebar ) {
+		$this->validate_sidebar( $sidebar );
+		$this->sidebar = $sidebar;
+	}
+
+	/**
+	 * Validates the sidebar.
+	 *
+	 * @since TBD
+	 *
+	 * @param callable|Settings_Sidebar $sidebar The sidebar to validate.
+	 *
+	 * @return void
+	 * @throws InvalidArgumentException If the sidebar is invalid.
+	 */
+	protected function validate_sidebar( $sidebar ) {
+		// If it's a callable or an instance of Settings_Sidebar, we're good.
+		if ( is_callable( $sidebar ) || $sidebar instanceof Settings_Sidebar ) {
+			return;
+		}
+
+		// Everything else is invalid.
+		throw new InvalidArgumentException(
```

Comment:
Should this function return a boolean just in case?
i.e. `if ( validate_sidebar() ) {...}`

Also - don't depend on the exception throwing a fatal - a catch block might handle it silently (although this code will still stop at the point we `throw`)

Reply:
> Should this function return a boolean just in case?

I don't think so. The reason for throwing in the first place is to ensure that we only have the data we expect in the class property, which is either a callback or `Settings_Sidebar` instance.

Where or whether the exception is caught is secondary. In fact, I would hope that we'd use a `catch` statement in an appropriate place to prevent an exception from bubbling up to the end user, if the circumstance warrants it.

For now, we'll be using it appropriately, and we can hope that any 3rd parties will discover any issues before publishing something.

If we really want to `catch` in the `add_sidebar()` method, I suppose we could do so, but I feel like it's a better development experience to surface errors in a way that is more difficult to ignore.